src/main/java/de/hpi/bpt/argos/common/ObservableImpl.java

```diff
@@ -1,45 +1,73 @@
 package de.hpi.bpt.argos.common;
 
-import java.util.HashSet;
-import java.util.Set;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.function.Consumer;
 
 /**
  * {@inheritDoc}
  * This is an abstract base implementation.
  */
-public abstract class ObservableImpl<Observer> implements Observable<Observer> {
+public class ObservableImpl<Observer> implements Observable<Observer> {
 
-	private Set<Observer> observers;
+	public enum ObserverOrder {
+		FIRST_IN_FIRST_OUT,
```

Comment:
I'm not sure if I like the name of those attributes. they suggest that the observers are removed in that order, not addressed.